packages/shared/src/components/drawers/NavDrawer.tsx

```diff
@@ -56,7 +62,11 @@ export function NavDrawer({
           <Button
             variant={ButtonVariant.Tertiary}
             size={ButtonSize.Small}
-            onClick={() => ref.current?.onClose()}
+            onClick={
+              shouldGoBack
+                ? () => router.push(user.permalink)
```

Comment:
What does the `NavDrawer` component have to do with user.permalink? 🤔 
Could it not be used for other origins?

Reply:
Ah, my mistake, yeah, this should have been in the other consuming component.